src/test/java/co/rsk/federate/bitcoin/BtcTransactionBuilder.java

```diff
@@ -0,0 +1,105 @@
+package co.rsk.federate.bitcoin;
+
+import co.rsk.bitcoinj.core.Address;
+import co.rsk.bitcoinj.core.BtcTransaction;
+import co.rsk.bitcoinj.core.Coin;
+import co.rsk.bitcoinj.core.NetworkParameters;
+import co.rsk.bitcoinj.core.TransactionInput;
+import co.rsk.bitcoinj.core.TransactionOutPoint;
+import co.rsk.bitcoinj.core.TransactionOutput;
+import co.rsk.bitcoinj.core.TransactionWitness;
+import co.rsk.bitcoinj.script.Script;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public final class BtcTransactionBuilder {
+
+    private final NetworkParameters networkParameters;
+    private final Map<Integer, TransactionWitness> transactionWitnesses = new HashMap<>();
+    private final List<TransactionInput> inputs = new ArrayList<>();
+    private final List<TransactionOutput> outputs = new ArrayList<>();
+
+    public BtcTransactionBuilder(NetworkParameters networkParameters) {
+        this.networkParameters = networkParameters;
+    }
+
+    public BtcTransactionBuilder withInput(TransactionInput transactionInput) {
+        inputs.add(transactionInput);
+        return this;
+    }
+
+    public BtcTransactionBuilder withInput(TransactionOutput transactionOutput) {
+        TransactionInput transactionInput = new TransactionInput(networkParameters, null,
+            new byte[]{}, transactionOutput.getOutPointFor(), transactionOutput.getValue());
+        inputs.add(transactionInput);
+        return this;
+    }
+
+    public BtcTransactionBuilder withWitness(int inputIndex,
+        TransactionWitness transactionWitness) {
+        transactionWitnesses.put(inputIndex, transactionWitness);
+        return this;
+    }
+
+    public BtcTransactionBuilder withOutput(Coin amount, Address address) {
+        outputs.add(
+            new TransactionOutput(networkParameters, null, amount, address)
+        );
+        return this;
+    }
+
+    public BtcTransaction build() {
+        BtcTransaction btcTransaction = new BtcTransaction(networkParameters);
+        addInputs(btcTransaction);
+        addOutputs(btcTransaction);
+        return btcTransaction;
+    }
+
+    private void addInputs(BtcTransaction btcTransaction) {
+        inputs.forEach(transactionInput -> {
+            int idx = btcTransaction.getInputs().size();
+            btcTransaction.addInput(transactionInput);
+            if (transactionWitnesses.containsKey(idx)) {
+                btcTransaction.setWitness(idx, transactionWitnesses.get(idx));
+            }
+        });
+    }
+
+    private void addOutputs(BtcTransaction btcTransaction) {
+        outputs.forEach(btcTransaction::addOutput);
+    }
+
+    public class InputBuilder {
+
+        private final Coin amount;
+        private Script scriptSig;
+        private int outpointIndex = 0;
+
+        public InputBuilder(Coin amount) {
```

Comment:
is it necessary for the amount to be part of the constructor?

Reply:
Well, it can be ZERO by default. It does not make much sense to me. I consider the field amount the main reason behind creating an input for testing. Most of the time, you will pass a value to the input. It would be very unusual for one to try to spend a ZERO value input; I even think it may not be possible.

Reply:
i think is better to have a default value (not zero), i think some times the important thing is just to have an input and the amount doesn't really matter in the context. Wdyt?

Reply:
It's ok to me. I'll add it.

Reply:
Done